PWGDQ/Tasks/dqEfficiency.cxx

```diff
@@ -803,6 +810,20 @@ struct AnalysisSameEventPairing {
     runMCGen(groupedMCTracks);
   }
 
+  void processJpsiToMuMuVertexingSkimmedFlat(soa::Filtered<MyEventsVtxCovSelected>::iterator const& event,
+                                             soa::Filtered<MyMuonTracksSelectedWithCov> const& muons,
+                                             ReducedMCEvents const& eventsMC, ReducedMCTracks const& tracksMC)
+  {
+    // Reset the fValues array
+    VarManager::ResetValues(0, VarManager::kNVars);
+    VarManager::FillEvent<gkEventFillMap>(event);
+    VarManager::FillEvent<gkMCEventFillMap>(event.reducedMCevent());
+
+    runPairing<VarManager::kJpsiToMuMu, gkEventFillMapWithCov, gkMCEventFillMap, gkMuonFillMapWithCov>(event, muons, muons, eventsMC, tracksMC, true);
+    auto groupedMCTracks = tracksMC.sliceBy(aod::reducedtrackMC::reducedMCeventId, event.reducedMCevent().globalIndex());
+    runMCGen(groupedMCTracks);
+  }
+
```

Comment:
This process function would not be needed, if you implement my comments from above.